tools/SourceKit/lib/SwiftLang/SwiftASTManager.cpp

```diff
@@ -197,7 +197,7 @@ namespace SourceKit {
         LOG_WARN_FUNC("did not find primary SourceFile");
         ConsumerRef->failed("did not find primary SourceFile");
       }
-    });
+    }, /*isStackDeep=*/true);
```

Comment:
This is unfortunate, it's going to be creating a new pthread every time to query the AST. Is there no way to make it work without making this change ?
I find strange that dispatch threads have limited stack also on Darwin but we did not have issues with querying the AST.

If there is no way around it I'd at least suggest to only have this change in effect when compiling for Windows.

Reply:
https://github.com/apple/swift-corelibs-libdispatch/blob/6c5b3ba86fce0480f702ba13fcb5f656af090b20/src/queue.c#L5175

This is hardcoded into libdispatch.  There is no way to adjust the stack size.  It is very likely that you are skimming past pretty close to the limit.

Reply:
That's my current theory. A [sample count of frame sizes](https://gist.github.com/gmittert/1d6691ba799da7ce923f0c7c75130b1c) seems to confirm that. Why Windows uses slightly more memory, I'm not sure yet.

Reply:
That stack trace looks to be from syntactic parsing, it is unrelated to `ASTUnit::Implementation::consumeAsync`.
Is the change inside `consumeAsync` really necessary or are the other changes making it redundant ?

Reply:
The consume async change is needed here. `"S:\b\swift\bin\sourcekitd-test.EXE" "-req=cursor" "-pos=9:25" "S:\swift\test\SourceKit\CursorInfo\rdar_30248264.swift" "--" "S:\swift\test\SourceKit\CursorInfo\rdar_30248264.swift"` stack overflows without it on windows. I'm currently trying to find a better way to do it. At worst, I'll `#if defined(_WIN32)` it, but I'd rather do something nicer.

The stack trace was more an example from another test that was stack overflowing showing that Windows typically does currently use more stack memory than Darwin.